src/connection/activate.ts

```diff
@@ -0,0 +1,88 @@
+import { sendAnalyticsEvent } from '@uniswap/analytics'
+import { InterfaceEventName, WalletConnectionResult } from '@uniswap/analytics-events'
+import { Connection } from 'connection/types'
+import { atom } from 'jotai'
+import { useAtomValue, useUpdateAtom } from 'jotai/utils'
+import { useCallback } from 'react'
+import { useAppDispatch } from 'state/hooks'
+import { updateSelectedWallet } from 'state/user/reducer'
+
+import { didUserReject } from './utils'
+
+export enum ActivationStatus {
+  PENDING,
+  ERROR,
+  IDLE,
+}
+
+type ActivationPendingState = { status: ActivationStatus.PENDING; connection: Connection }
+type ActivationErrorState = { status: ActivationStatus.ERROR; connection: Connection; error: any }
+const IDLE_ACTIVATION_STATE = { status: ActivationStatus.IDLE } as const
+type ActivationState = ActivationPendingState | ActivationErrorState | typeof IDLE_ACTIVATION_STATE
+
+const activationStateAtom = atom<ActivationState>(IDLE_ACTIVATION_STATE)
+
+function useTryActivation() {
+  const dispatch = useAppDispatch()
+  const setActivationState = useUpdateAtom(activationStateAtom)
+
+  return useCallback(
+    async (connection: Connection, onSuccess: () => void) => {
+      /*
+       * Skips wallet connection if the connection should override the default
+       * behavior, i.e. install MetaMask or launch Coinbase app
+       */
+      if (connection.overrideActivate?.()) return
+
+      try {
+        setActivationState({ status: ActivationStatus.PENDING, connection })
+        await connection.connector.activate()
+
+        console.debug(`Connection activated: ${connection.getName()}`)
```

Comment:
Additional logging would make any future issues easier to debug:

```suggestion

        console.debug(`Connection activating: ${connection.getName()}`)
        await connection.connector.activate()
        console.debug(`Connection activated: ${connection.getName()}`)
```